trl/trainer/kto_trainer.py

```diff
@@ -312,25 +312,50 @@ def make_inputs_require_grad(module, input, output):
         self.desirable_weight = args.desirable_weight
         self.undesirable_weight = args.undesirable_weight
 
-        # tokenize the dataset
-        columns_to_remove = [col for col in train_dataset.column_names if col not in ["prompt", "completion", "label"]]
-        train_dataset = train_dataset.map(
-            self.shuffle_completion, batched=True, batch_size=128, remove_columns=columns_to_remove
-        )
-        train_dataset = train_dataset.map(self.tokenize_row)
+        # get KL datasets
+        train_KL_dataset = train_dataset.map(self.get_KL_dataset, batched=True, batch_size=1000)
+        if eval_dataset is not None:
+            eval_KL_dataset = eval_dataset.map(self.get_KL_dataset, batched=True, batch_size=1000)
 
+        # tokenize the datasets
+        train_dataset = train_dataset.map(lambda row: self.tokenize_row(row, prefix=""), remove_columns=train_dataset.column_names)
+        train_KL_dataset = train_KL_dataset.map(lambda row: self.tokenize_row(row, prefix="KL_"), remove_columns=train_KL_dataset.column_names)
+        
         if eval_dataset is not None:
-            columns_to_remove = [
-                col for col in eval_dataset.column_names if col not in ["prompt", "completion", "label"]
-            ]
-            eval_dataset = eval_dataset.map(
-                self.shuffle_completion, batched=True, batch_size=128, remove_columns=columns_to_remove
-            )
-            eval_dataset = eval_dataset.map(self.tokenize_row)
+            eval_dataset = eval_dataset.map(lambda row: self.tokenize_row(row, prefix=""), remove_columns=eval_dataset.column_names)
+            eval_KL_dataset = eval_KL_dataset.map(lambda row: self.tokenize_row(row, prefix="KL_"), remove_columns=eval_KL_dataset.column_names)
+
+        # merge the datasets
+        train_dataset = concatenate_datasets([train_dataset, train_KL_dataset], axis=1)
+        eval_dataset = concatenate_datasets([eval_dataset, eval_KL_dataset], axis=1)
+
+        desirable = train_dataset.filter(lambda x: x["label"])
+        undesirable = train_dataset.filter(lambda x: not x["label"])
+
+        if (len(desirable) != len(undesirable)):
+            des_weight_lower_bound = (len(undesirable) * self.undesirable_weight / len(desirable)) * 1
```

Comment:
Let's add this for clarity:

```suggestion
            # The lower and upper bounds come from Eq. (8) of https://arxiv.org/abs/2402.01306
            des_weight_lower_bound = (len(undesirable) * self.undesirable_weight / len(desirable)) * 1
```